tests/qiskit_convert_test.py

```diff
@@ -1165,6 +1166,23 @@ def test_symbolic_param_conv() -> None:
         }
     )
 
+def test_implicit_swap_warning() -> None:
+    c = Circuit(2).H(0).SWAP(0, 1)
+    c.replace_SWAPs()
+    with pytest.warns(UserWarning, match="The pytket Circuit contains implicit qubit"):
+        tk_to_qiskit(c)
+
+    state_backend = AerStateBackend()
```

Comment:
I wrote this test without the flag added to ensure that it failed. 

The warning only appears for `AerBackend` and not `AerStateBackend`. I would've expected both to generate a warning (i.e. `record` to be of length 2)

Reply:
Ah its because the state and unitary backends replace implicit swaps with SWAP gates whereas `AerBackend` does not.